Fix/2021.3 integration tests

jwillebrands opened the pull request:
# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [x] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Issue number: https://github.com/bazelbuild/intellij/issues/3001

# Description of this change

- Replaced annotator test code based on deprecated `AnnotationHolderImpl` with the `CodeInsightTestUtil.testAnnotator` utility method.
- Replaced deprecated use of `AnnotationHolder.createXAnnotation` with the new `AnnotationBuilder` based API.
- Run tests with `idea.force.use.core.classloader` set to `true`. Solution proposed by IntelliJ [here](https://plugins.jetbrains.com/docs/intellij/api-changes-list-2021.html), and a bit more context [here](https://github.com/JetBrains/gradle-intellij-plugin/blob/0873e0418a2bd01e8fa74c98811af1a567cdbced/src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt#L812).

alice-ks commented on testing/src/com/google/idea/testing/ServiceHelper.java:
```diff
@@ -141,8 +142,6 @@
-        () ->
-            ((MutablePicoContainer) componentManager.getPicoContainer())
-                .unregisterComponent(key.getName()));
+        () -> new ComponentManagerWrapper(componentManager).unregisterComponent(key.getName()));
```
Even though this can be implemented as a Wrapper, it's not directly necessary. It would also be possible to introduce a static method to `BaseSdkCompat` which takes `componentManager` and `key.getName()` as method parameters. Going for a method in `BaseSdkCompat` is more light-weight and hence preferred. Can you please switch to that approach?

jwillebrands replied:
Sure, will do. Just double-checking: I'm assuming this should go in `BaseSdkTestCompat` then. Is that correct?

alice-ks replied:
Right. Since it's only used from test code, it should go into `BaseSdkTestCompat`. Thanks for double-checking.

alice-ks commented on testing/testcompat/v203/com/google/idea/sdkcompat/ComponentManagerWrapper.java:
```diff
@@ -0,0 +1,19 @@
+
+import com.intellij.openapi.components.ComponentManager;
+
+// #api212
```
Here and for the other classes: Please add a short explanation on what to do when addressing the #api212 tag in the future.

alice-ks reviewed (changes requested):
Thank you for the adjustments for 2021.3 and figuring out how to fix the tests. The IntelliJ CE 2021.3 tests are green now. The IntelliJ UE 2021.3 tests still fail, but I guess that requires additional adjustments. Let's get this PR in to build on top of it.

I attached some comments. Please take a look. They should be quick to address. I'll do the next review round much more quickly (sorry that the initial review took a bit of time).

jwillebrands commented on base/src/com/google/idea/blaze/base/lang/buildfile/validation/HighlightingAnnotator.java:
```diff
@@ -22,38 +22,46 @@
 
-/** Additional syntax highlighting, based on parsed PSI elements. */
+/**
+ * Additional syntax highlighting, based on parsed PSI elements.
```
Sorry, I couldn't get the formatter to play along. Thought I had omitted all of the mangled comments.

jwillebrands commented on testing/testcompat/v212/com/google/idea/sdkcompat/BaseSdkTestCompat.java:
```diff
@@ -15,10 +15,29 @@
+    AnnotationHolderImpl holder = new AnnotationHolderImpl(new AnnotationSession(file));
+    for (PsiElement element : elements) {
+      holder.runAnnotatorWithContext(element, annotator);
+    }
```
Yeah, that makes sense. Especially having also refactored the annotation creation to using the Builder.

jwillebrands commented:
Thanks for the review. No worries about taking a bit of time, I was still investigating the last failure. Agreed that we should move that to a separate PR.

I've addressed your comments. Should be good now. Thanks!

alice-ks commented:
Great. Thanks. I'll start the import of the changes.
